### ISSUE TITLE
Error on __schema query with comments

### ISSUE THREAD

**Issue by ovidiubuligan** [graphql-dotnet/parser#80]
I am using the latest `3.0.0-preview-1631` so parser version is  `5.1.2` . The following query gives an error :

```javascript
{
  __schema {
    
    types {
      kind
      
    }
  }
  
# __schema{
#   types {
#     name
#   }
# }
}
```

The error is :  
```javascript
{
  "errors": [
    {
      "message": "GraphQL.ExecutionError: ParserContext has 1 not applied comments.\r\n ---> System.ApplicationException: ParserContext has 1 not applied comments.\r\n   at GraphQLParser.ParserContext.Dispose()\r\n   at GraphQLParser.Parser.Parse(ISource source)\r\n   at GraphQL.Execution.GraphQLDocumentBuilder.Build(String body) in C:\\projects\\graphql-dotnet\\src\\GraphQL\\Execution\\GraphQLDocumentBuilder.cs:line 20\r\n   at GraphQL.DocumentExecuter.ExecuteAsync(ExecutionOptions options) in C:\\projects\\graphql-dotnet\\src\\GraphQL\\Execution\\DocumentExecuter.cs:line 86\r\n   --- End of inner exception stack trace ---",
      "extensions": {
        "code": "APPLICATION",
        "codes": [
          "APPLICATION"
        ]
      }
    }
  ],
  "extensions": {
    "tracing": {
      "version": 1,
      "startTime": "2020-07-27T09:02:21.0125943Z",
      "endTime": "2020-07-27T09:02:21.0125943Z",
      "duration": 354200,
      "parsing": {
        "startOffset": 5200,
        "duration": 303400
      },
      "validation": {
        "startOffset": 0,
        "duration": 0
      },
      "execution": {
        "resolvers": []
      }
    }
  }
}
```

**Comment by sungam3r** [comment]
Relates to #33 and #69.

**Comment by rmeshksar** [comment]
Hi, Is there any update on the fix for this issue?

**Comment by sungam3r** [comment]
Nope

**Comment by rmeshksar** [comment]
I thought the following issue has been fixed:
https://github.com/graphql-dotnet/graphql-dotnet/issues/1415

It only happens if the comment does not describe any field or selection or block, like having comment after the query or before closing } oft he query.

**Comment by Shane32** [comment]
This also produces incorrect results:
```gql
query _ {
    person {
        name
        #comment1
    }
    #comment2
    test {
        alt
    }
}
```
What happens is that `comment1` is incorrectly applied to `alt`.

**Comment by sungam3r** [comment]
Fixed in #104.
